scipy/spatial/spherical_voronoi.py

```diff
@@ -0,0 +1,416 @@
+"""
+Spherical Voronoi Code
+
+.. versionadded:: 0.17.0
+
+"""
+#
+# Copyright (C)  Tyler Reddy, Ross Hemsley, Edd Edmondson, Nikolai Nowaczyk, Joe Pitt-Francis, 2015.
+#
+# Distributed under the same BSD license as Scipy.
+#
+
+import numpy as np
+import scipy
+import math
+
+__all__ = ['SphericalVoronoi']
+
+def calc_circumcenter_circumsphere_tetrahedron_vectorized(tetrahedron_coord_array):
+    '''An alternative implementation based on http://mathworld.wolfram.com/Circumsphere.html because of issues with the initial implementation from the Berkeley page.
+    Vectorized version for use with multiple tetrahedra in tetrahedron_coord_array -- the latter should have shape (N, 4, 3).'''
+    num_tetrahedra = tetrahedron_coord_array.shape[0]
+    #reshape the tetrahedron_coord_array to place all tetrahedra consecutively without nesting
+    tetrahedron_coord_array = np.reshape(tetrahedron_coord_array, (tetrahedron_coord_array.shape[0] * tetrahedron_coord_array.shape[1], tetrahedron_coord_array.shape[2]))
```

Comment:
Your style is much more verbose than the rest of scipy, which will make it hard to conform to line-length limits and (IMO) reduces readability. I'd recommend dropping nonessential information from your variable names. For example, pretty much every instance of the `array_foo_values` pattern could be more succinctly named `foo`.